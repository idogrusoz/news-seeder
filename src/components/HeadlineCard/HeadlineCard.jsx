import React from 'react';

//Displays top headline. Takes an article object as props and displays the basic data
const HeadlineCard = ({ article }) => {
    const { urlToImage, title, source, description } = article;
    const styles = {
        wrapper: {
            width: '300px',
            margin: '20px',
            padding: '10px',
            cursor: 'pointer',
            fontSize: '15px',
        },
        h2: {
            margin: '0 auto',
            fontWeight: 400,
        },
        h5: {
            margin: '5px',
            textAlign: 'center',
            fontWeight: 600,
            fontSize: '12px',
        },
        image: {
            width: '300px',
            height: '172px',
            objectFit: 'cover',
        },
        p: {
            textAlign: 'left',
            textIndent: '2em',
            margin: '0 2px',
            color: 'grey',
            fontSize: '14px',
        },
    };

    return (
        <div style={styles.wrapper}>
            <img src={urlToImage || process.env.PUBLIC_URL + '/social-seeder.jpeg'} alt={title} style={styles.image} />
            <h3 style={styles.h2}>{title}</h3>
            <h5 style={styles.h5}>Source: {source.name}</h5>
            <p style={styles.p}>{description}</p>
        </div>
    );
};

export default HeadlineCard;
