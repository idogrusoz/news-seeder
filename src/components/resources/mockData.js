import { act } from '@testing-library/react';

export let mockNews = [
    {
        source: {
            id: null,
            name: 'YouTube',
        },
        author: null,
        title: 'mock-title',
        description: 'mock-description',
        url: 'https://www.mock-url.be',
        urlToImage: 'https://mock-url.be/maxresdefault.jpg',
        publishedAt: '2020-12-21T08:00:30Z',
        content: null,
    },
    {
        source: {
            id: null,
            name: 'YouTube',
        },
        author: null,
        title: 'mock-title',
        description: 'mock-description',
        url: 'https://www.mock-url2.be',
        urlToImage: null,
        publishedAt: '2020-12-21T08:00:30Z',
        content: 'some text <script>danger</script> [+42]',
    },
];

export const waitForComponentToPaint = async (wrapper) => {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, 0));
        wrapper.update();
    });
};
