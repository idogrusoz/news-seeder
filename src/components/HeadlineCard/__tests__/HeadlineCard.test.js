import { shallow } from 'enzyme';
import HeadlineCard from '../HeadlineCard';
import { mockNews } from '../../resources/mockData';

const mockSet = jest.fn();
const mockHistory = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: () => mockHistory(),
    }),
}));

describe('HeadlineCard test', () => {
    const mock = mockNews[0];
    const mockWithNullImage = mockNews[1];
    let wrapper = shallow(<HeadlineCard article={mock} setSelected={mockSet} />);
    it('redirects to article when clicked', () => {
        wrapper.simulate('click');
        expect(mockSet).toBeCalledWith(mock);
        expect(mockHistory).toBeCalled();
    });
    it('displays image', () => {
        const image = wrapper.find('img');
        expect(image.prop('src')).toEqual(mock.urlToImage);
    });
    it('displays the title', () => {
        const title = wrapper.find('h3');
        expect(title.text()).toEqual(mock.title);
    });
    it('displays social seeder logo when image is null', () => {
        wrapper.unmount();
        mockNews.urlToImage = null;
        wrapper = shallow(<HeadlineCard article={mockWithNullImage} />);
        expect(wrapper.find('img').prop('src')).toEqual(process.env.PUBLIC_URL + '/social-seeder.jpeg');
    });
});
