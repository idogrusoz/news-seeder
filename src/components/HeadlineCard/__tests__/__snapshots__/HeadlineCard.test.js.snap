// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadlineCard test matches the snapshot 1`] = `ShallowWrapper {}`;
